File: simpson.js
function simpson(a, b, n, av) {
	h = (b - a) / n;
	xarr = [b];
    yarr = [b];
    xarr[0] = a;
    xarr[b-1] = b;

    for (let i = 1; i < b-1; i++) {
    	xarr[i] = xarr[0] + i*h;
    }

    console.log(xarr);

    for (let i = 0; i < b; i++) {
    	yarr[i] = horner(av, xarr[i]);
    }

    console.log(yarr);

    simres = yarr[0];

    for (let i = 1; i < (b-1); i++) {
    	simres += (i&1 ? 4 : 2) * yarr[i];
    }

    simres += yarr[b-1];
    return (h/3)*simres;
}

function horner(a, x) {
	result = a[0];
	for (let i = 1; i < a.length; i++) {
		result = result * x + a[i];
	}
	return result;
}

console.log(simpson(1, 5, 4, [1, 3, 2]))

File: simpson.m
function[w] = simpson(a, b, n, av)
    h = (b - a) / n;
    
    xarr = [b];
    yarr = [b];
    xarr(1) = a;
    xarr(b) = b;
    
    for i=2:(b-1)
       xarr(i) = xarr(1) + (i-1)*h;
    end
    
    for i=1:b
       yarr(i) = horner(av, xarr(i)); 
    end
    
    simres = yarr(1);
    
    for i=2:(b-1)
        if mod(i, 2) == 0
            simres = simres + 4 * yarr(i);
        else
            simres = simres + 2 * yarr(i);
        end
    end
    
    simres = simres + yarr(b);
    w = (h / 3) * simres;
end

% 2.6x^2 * lnx [1,2; 2,2] n=10